presentation-compiler/src/main/dotty/tools/pc/RawScalaPresentationCompiler.scala

```diff
@@ -0,0 +1,340 @@
+package dotty.tools.pc
+
+import java.io.File
+import java.net.URI
+import java.nio.file.Path
+import java.util.Optional
+import java.util as ju
+
+import scala.jdk.CollectionConverters._
+import scala.language.unsafeNulls
+import scala.meta.internal.metals.CompilerVirtualFileParams
+import scala.meta.pc.reports.EmptyReportContext
+import scala.meta.internal.metals.PcQueryContext
+import scala.meta.pc.reports.ReportContext
+import scala.meta.internal.metals.ReportLevel
+import scala.meta.internal.mtags.CommonMtagsEnrichments.*
+import scala.meta.internal.pc.EmptySymbolSearch
+import scala.meta.internal.pc.PresentationCompilerConfigImpl
+import scala.meta.pc.*
+import scala.meta.pc.{PcSymbolInformation as IPcSymbolInformation}
+
+import dotty.tools.pc.completions.CompletionProvider
+import dotty.tools.pc.InferExpectedType
+import dotty.tools.pc.completions.OverrideCompletions
+import dotty.tools.pc.buildinfo.BuildInfo
+import dotty.tools.pc.SymbolInformationProvider
+import dotty.tools.dotc.interactive.InteractiveDriver
+
+import org.eclipse.lsp4j.DocumentHighlight
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.lsp4j as l
+
+
+case class RawScalaPresentationCompiler(
+    buildTargetIdentifier: String = "",
+    buildTargetName: Option[String] = None,
+    classpath: Seq[Path] = Nil,
+    options: List[String] = Nil,
+    search: SymbolSearch = EmptySymbolSearch,
+    config: PresentationCompilerConfig = PresentationCompilerConfigImpl(),
+    folderPath: Option[Path] = None,
+    reportsLevel: ReportLevel = ReportLevel.Info,
+    completionItemPriority: CompletionItemPriority = (_: String) => 0,
+    reportContext: ReportContext = EmptyReportContext()
+) extends RawPresentationCompiler:
+
+  def this() = this("uninitialized-presentation-compiler")
+
+  given ReportContext = reportContext
+
+  override def supportedCodeActions(): ju.List[String] = List(
+     CodeActionId.ConvertToNamedArguments,
+     CodeActionId.ImplementAbstractMembers,
+     CodeActionId.ExtractMethod,
+     CodeActionId.InlineValue,
+     CodeActionId.InsertInferredType,
+     CodeActionId.InsertInferredMethod,
+     PcConvertToNamedLambdaParameters.codeActionId
+   ).asJava
+
+  override val scalaVersion = BuildInfo.scalaVersion
+
+  private val forbiddenOptions = Set("-print-lines", "-print-tasty")
+  private val forbiddenDoubleOptions = Set.empty[String]
+
+  val driverSettings =
```

Comment:
This seems duplicated with the Scala Presentation Compiler, shouldn't we reuse this class in the old ScalaPResentationCompiler?